File: src/SignalRServiceExtension/Bindings/SignalRGroupAction.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.Runtime.Serialization;

namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
{
    [JsonObject]
    public class SignalRGroupAction
    {
        [JsonProperty("connectionId")]
        public string ConnectionId { get; set; }
        [JsonProperty("userId")]
        public string UserId { get; set; }
        [JsonProperty("groupName"), JsonRequired]
        public string GroupName { get; set; }
        [JsonProperty("action"), JsonRequired]
        public GroupAction Action { get; set; }
    }

    [JsonConverter(typeof(StringEnumConverter))]
    public enum GroupAction
    {
        [EnumMember(Value = "add")]
        Add,
        [EnumMember(Value = "remove")]
        Remove
    }
}

File: binding-library/java/src/main/java/com/microsoft/azure/functions/signalr/SignalRGroupAction.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */
package com.microsoft.azure.functions.signalr;

/**
 * <p>
 * SignalR group action (used with SignalR output binding)
 * </p>
 *
 * @since 1.0.0
 */
public class SignalRGroupAction {
    /**
     * User to add to or remove from group
     */
    public String userId = "";

    /**
     * Group to add user to or remove user from
     */
    public String groupName = "";

    /**
     * Action to take ("add" or "remove")
     */
    public String action = "";
}